src/Compilers/Core/Portable/Symbols/ISymbolExtensions_IsAccessibleWithin.cs

```diff
@@ -0,0 +1,347 @@
+
+            bool inheritsFromIgnoringConstruction(ITypeSymbol type, INamedTypeSymbol baseType)
+            {
+                Debug.Assert(type.IsDefinition);
+                Debug.Assert(baseType.IsDefinition);
+
+                for (var current = type; current != null; current = current.BaseType?.OriginalDefinition)
+                {
+                    if (sameOriginalNamedType(baseType, current as INamedTypeSymbol))
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            bool sameOriginalNamedType(INamedTypeSymbol t1, INamedTypeSymbol t2)
+            {
+                Debug.Assert(t1 != null);
+                Debug.Assert(t1.IsDefinition);
+                Debug.Assert(t2?.IsDefinition != false);
+
+                // We expect a given named type definition to satisfy reference identity.
+                // But we relax this to permit a type to be represented by multiple symbols (e.g. separate compilations).
+                // Note that the same symbol is expected to have the identical name as itself, despite VB language rules.
+                return t1 == t2 || t2 != null && t1.MetadataName == t2.MetadataName && t1.Arity == t2.Arity && sameOriginalSymbol(t1.ContainingSymbol, t2.ContainingSymbol);
+
+                bool sameOriginalSymbol(ISymbol s1, ISymbol s2)
+                {
+                    if (s1 == s2)
+                    {
+                        return true;
+                    }
+
+                    if (s1 == null || s2 == null)
+                    {
+                        return false;
+                    }
+
+                    switch (s1.Kind)
+                    {
+                        case SymbolKind.NamedType:
+                            return sameOriginalNamedType(s1 as INamedTypeSymbol, s2 as INamedTypeSymbol);
+                        case SymbolKind.Namespace:
+                            return sameNamespace(s1 as INamespaceSymbol, s2 as INamespaceSymbol);
+                        case SymbolKind.Assembly:
+                            return sameAssembly(s1 as IAssemblySymbol, s2 as IAssemblySymbol);
+                        case SymbolKind.NetModule:
+                            return sameModule(s1 as IModuleSymbol, s2 as IModuleSymbol);
+                        default:
+                            return false;
+                    }
+                }
+
+                bool sameModule(IModuleSymbol m1, IModuleSymbol m2)
+                {
+                    Debug.Assert(m1 != null);
+                    // We don't need to check the module name, as modules are effectively merged in the containing assembly.
+                    return m2 != null && sameOriginalSymbol(m1.ContainingSymbol, m2.ContainingSymbol);
+                }
+
+                bool sameAssembly(IAssemblySymbol a1, IAssemblySymbol a2)
+                {
+                    return a1.Equals(a2);
```

Comment:
>a1.Equals(a2) [](start = 27, length = 13)

I believe, this code will not give accurate answers across languages

Reply:
I'll file a bug on the compilers.

---
In reply to: [185312547](https://github.com/dotnet/roslyn/pull/26367#discussion_r185312547) [](ancestors = 185312547)

Reply:
> I'll file a bug on the compilers.

This is definitely not a bug. This is by design

---
In reply to: [185313035](https://github.com/dotnet/roslyn/pull/26367#discussion_r185313035) [](ancestors = 185313035,185312547)

Reply:
Filed as #26542.

---
In reply to: [185313534](https://github.com/dotnet/roslyn/pull/26367#discussion_r185313534) [](ancestors = 185313534,185313035,185312547)